### DISCUSSION TITLE
Nextcloud making WOPI request using incorrect IP.

### DISCUSSION THREAD

**Discussion by madsamjp** [nextcloud/all-in-one#2176] [Questions] [answered]
I have a Nextcloud all-in-one instance running on my home server. I access this using nginx-proxy-manager hosted in a VPS with a Wireguard VPN to my docker VM. When I try and open a Collabora document, it fails. This guide gives some indication as to what the issue might be: https://github.com/nextcloud/all-in-one/discussions/1358?sort=new

When I check the logs, it says 
`WOPI request denied from XX.XX.XX.XX as it does not match the configured ranges: ......`

The IP address above that was denied is my home ip address given by my ISP. I could whitelist this address, however that is not ideal as my ISP regularly switches my IP address.

I cant seem to figure out how Nextcloud gets my home IP address when I access it through a VPS/VPN. Surely it should be using either the VPS IP or an IP from a docker network within my system. 

Basically, my question is, how do I make nextcloud use my reverse proxy host (VPS) IP address when trying to open a Collabora document instead of my home IP address?

**Comment by szaimen** [comment, marked as answer]
Hi, so for your setup the best workaround seems to be to add ,0.0.0.0/0 permanently to the Allow list for WOPI requests.

Apart from that I would recommend to create a feature request to allow to add domains into this filed which would then automatically use the correct ip-addresss that was added to the given domain: https://github.com/nextcloud/richdocuments/issues.

Apart from that would a JWT-secret solve all these annoying details as then the wopi-allowlist would not be needed anymore. But this would also be a feature request for https://github.com/nextcloud/richdocuments/issues.

**Reply by meonkeys** [reply]
CODE does support using hostnames/domain names in the WOPI host allow list: https://sdk.collaboraonline.com/docs/installation/CODE_Docker_image.html#setting-the-application-configuration-dynamically-via-environment-variables
